openmsupply/open-msupply
# 1378 programs sorting out statuses and labels for program patients and encounters

Fixes #1378

# 👩🏻‍💻 What does this PR do? 
- Add status to program table so that it could be filtered and/or be displayed in ListViews.
- Rename `Encounter Status` to `Additional Information`
- Save testing result to program_events table
- Disable `Add Encounter` button if the patient has no active programs
- Encounter Search List will only display encounters for programs that are active
- Added missed status to Encounter status enum and configured any encounters that are in the past and haven't been completed to be `Missed`

# 🧪 How has/should this change been tested? 
- Go to encounter views to see rename.
- For Care Program status change: 
      - Add a care program or go to the program view and click a care program
      - Change the status of said program
      - See status reflected in the list view
      - Change status to anything other than Active... see that 'Add Encounter' button has been disabled if that was the patient's only active program
- For missed encounters: Create an encounter in the past and see it has been marked as missed
- For result: Change the result of an HIV test encounter to anything and see the change in program_events table

## 💌 Any notes for the reviewer?
- We decided that the `Died` program status shouldn't change fields in other forms, but should it be added to ProgramEnrolmentStatus? 
- Not sure if other program enrolment forms should have the option to change status or if the enrolment status should be optional.
- UI Changes for additional info column will be done in #1270
